crates/extensions/c8y_auth_proxy/src/actor.rs

```diff
@@ -30,30 +37,59 @@ pub struct C8yAuthProxyBuilder {
     bind_port: u16,
     signal_sender: mpsc::Sender<RuntimeRequest>,
     signal_receiver: mpsc::Receiver<RuntimeRequest>,
+    server_certificate: Vec<Vec<u8>>,
+    server_private_key: Vec<u8>,
 }
 
 impl C8yAuthProxyBuilder {
     pub fn try_from_config(
         config: &TEdgeConfig,
         jwt: &mut ServerActorBuilder<C8YJwtRetriever, Sequential>,
-    ) -> Result<Self, ConfigNotSet> {
+    ) -> anyhow::Result<Self> {
         let app_state = AppState {
             target_host: format!("https://{}", config.c8y.http.or_config_not_set()?).into(),
             token_manager: TokenManager::new(JwtRetriever::new("C8Y-PROXY => JWT", jwt)).shared(),
         };
         let bind = &config.c8y.proxy.bind;
         let (signal_sender, signal_receiver) = mpsc::channel(10);
+        let (server_certificate, server_private_key) = load_certificate_and_key(config)?;
 
         Ok(Self {
             app_state,
             bind_address: bind.address,
             bind_port: bind.port,
             signal_sender,
             signal_receiver,
+            server_certificate,
+            server_private_key,
         })
     }
 }
 
+fn load_certificate_and_key(config: &TEdgeConfig) -> anyhow::Result<(Vec<Vec<u8>>, Vec<u8>)> {
+    let paths = tedge_config_macros::all_or_nothing((
+        config.c8y.proxy.cert_path.as_ref(),
+        config.c8y.proxy.key_path.as_ref(),
+    ))
+    .map_err(|e| anyhow!("{e}"))?;
+
+    match paths {
+        Some((external_cert_file, external_key_file)) => Ok((
+            load_cert(external_cert_file)
+                .with_context(|| format!("reading certificate configured in {C8yProxyCertPath}"))?,
+            load_pkey(external_key_file).with_context(|| {
+                format!("reading private key configured in `{C8yProxyKeyPath}`")
+            })?,
+        )),
+        None => Ok((
+            load_cert(&config.device.cert_path)
+                .with_context(|| format!("reading certificate configured in `{DeviceCertPath}`"))?,
+            load_pkey(&config.device.key_path)
+                .with_context(|| format!("reading private key configured in `{DeviceKeyPath}`"))?,
```

Comment:
The `c8y_auth_proxy` should not fall-back to these device certificates if no certificates have been specifically provided for `c8y.proxy.cert_path`. Indeed, the former are used to authenticate the device over MQTT and are owned by mosquitto, while the mapper is running as `tedge`.

If no certificate is provided, I would simply let  `c8y_auth_proxy` expose an HTTP service.

This might be the root cause of all the system test failures.